pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -593,29 +596,77 @@ private static Expression toExpression(SqlNode node) {
           // Move on to process default logic.
         }
       default:
-        SqlBasicCall funcSqlNode = (SqlBasicCall) node;
-        String funcName = funcSqlNode.getOperator().getKind().name();
-        if (funcSqlNode.getOperator().getKind() == SqlKind.OTHER_FUNCTION) {
-          funcName = funcSqlNode.getOperator().getName();
+        return evaluateFunctionExpression((SqlBasicCall) node);
+    }
+  }
+
+  private static String extractFunctionName(SqlBasicCall funcSqlNode) {
+    String funcName = funcSqlNode.getOperator().getKind().name();
+    if (funcSqlNode.getOperator().getKind() == SqlKind.OTHER_FUNCTION) {
+      funcName = funcSqlNode.getOperator().getName();
+    }
+    if (funcName.equalsIgnoreCase(SqlKind.COUNT.toString()) && (funcSqlNode.getFunctionQuantifier() != null)
```

Comment:
i didn't quite follow why there is special casing for  DISTINCTCOUNT ?

Reply:
this is a special handling for the case of `COUNT(DISTINCT A)`.